### PR TITLE
Fix kappa

### PR DESCRIPTION
**Pull request by AakashKumarNain**
This PR fixes the problems mentioned in #906 

Highlights:

1. Works with/without keras model.
2. Can handle sparse labels for multi-class classification.
3. Can handle the cases where user want to treat the problem as a regression problem

### CODE REVIEW

**Comment by AakashKumarNain** [comment]
@facaiy @seanpmorgan @WindQAQ This is the best I could think of to keep things as simple and clean as possible. But if you feel we need to change something, I am open to suggestions.

**Comment by AakashKumarNain** [comment]
@googlebot I fixed it.

**Comment by AakashKumarNain** [comment]
@googlebot I fixed it.

File: tensorflow_addons/metrics/cohens_kappa.py [hunk 1]
```diff
@@ -67,18 +67,26 @@ def __init__(self,
...
+          weightage: (optional) Weighting to be considered for calculating
             kappa statistics. A valid value is one of
-            [None, 'linear', 'quadratic']. Defaults to `None`.
-          dtype: (Optional) Data type of the metric result.
-            Defaults to `None`.
+            [None, 'linear', 'quadratic']. Defaults to `None`
+          sparse_lables: (bool) Valid only for multi-class scenario.
+            If True, ground truth labels are expected tp be integers
+            and not one-hot encoded
+          reg_values: (bool) If set, that means the problem is being treated
```
Comment (seanpmorgan) [inline tensorflow_addons/metrics/cohens_kappa.py]: nit: Could we make this `regression (bool)` . `reg_values` seems a bit underspecified since `reg` could mean regression, regularization, etc. Granted there wouldn't be regularization in this metric but its in the vocabulary.

File: tensorflow_addons/metrics/cohens_kappa.py [hunk 2]
```diff
@@ -89,8 +97,18 @@ def __init__(self,
         if weightage not in (None, 'linear', 'quadratic'):
             raise ValueError("Unknown kappa weighting type.")
 
+        if num_classes == 2:
+            self._update = self._update_binary_class_model
```
Comment (seanpmorgan) [inline tensorflow_addons/metrics/cohens_kappa.py]: Could you add a test case for `categorical_crossentropy` on a two class problem? I believe this will have issues.

File: tensorflow_addons/metrics/cohens_kappa.py (hunk 1 above)
Comment (AakashKumarNain) [reply tensorflow_addons/metrics/cohens_kappa.py]: Makes sense

File: tensorflow_addons/metrics/cohens_kappa.py (hunk 2 above)
Comment (AakashKumarNain) [reply tensorflow_addons/metrics/cohens_kappa.py]: You want test cases for binary case, right?

**Comment by AakashKumarNain** [comment]
Thanks @seanpmorgan for the feedback. I have renamed that variable and added more tests to cover all the cases.

File: tensorflow_addons/metrics/cohens_kappa.py (hunk 2 above)
Comment (AakashKumarNain) [reply tensorflow_addons/metrics/cohens_kappa.py]: Done. Please check

**Comment by seanpmorgan** [comment]
We'll add this to 0.8.2 patch release

**Comment by AakashKumarNain** [comment]
Cool. Thanks for the valuable feedback
